docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,146 @@
+
+  return <Channel channel={activeChannel}>{children}</Channel>;
+};
+```
+
+Example with `setActiveChannel` function:
+
+```tsx
+import { useEffect } from 'react';
+import { useCreateChatClient, useChatContext, Chat, Channel } from 'stream-chat-react';
+
+const ActiveChannelSetter = ({ channelId, channelType }) => {
+  const { client, setActiveChannel } = useChatContext();
+
+  useEffect(() => {
+    const channel = client.channel(channelId, channelType);
+    setActiveChannel(channel);
+  }, [channelType, channelId]);
+
+  return null;
+};
+
+const App = () => {
+  const client = useCreateChatClient(userData);
+
+  if (!client) return <div>Loading...</div>;
+
+  return (
+    <Chat client={client}>
+      <ActiveChannelSetter channelId='random' channelType='messaging' />
+      <Channel>{/** ...other components... */}</Channel>
+    </Chat>
+  );
+};
+```
+
+Afterwards - to access currently active channel you can use `useChatContext` hook as well from any component (or custom hook) which is either direct or indirect child of a Chat component:
+
+```tsx
+import { useChatContext } from 'stream-chat-react';
+
+export const useActiveChannel = () => {
+  const { channel } = useChatContext();
+
+  return channel;
+};
+```
+
+Currently active [channel state](https://github.com/GetStream/stream-chat-react/blob/master/src/context/ChannelStateContext.tsx#L36-L56) can be accessed through the [`ChannelStateContext`](../components/contexts/channel-state-context.mdx) with the help of the `useChannelStateContext` hook - meaning any component which is either direct or indirect child of the `Channel` component can access such state.
+
+The example bellow shows how to reach `members` property coming from the channel state:
+
+```tsx
+import { Channel, useChannelStateContext } from 'stream-chat-react';
+
+const MembersCount = () => {
+  const { members } = useChannelStateContext();
+
+  return <div>{Object.keys(members).length}</div>;
+};
+
+const ChannelWrapper = () => (
+  <Channel>
+    <MembersCount />
+  </Channel>
+);
+```
+
+## Channel List State
+
+[`ChannelList`](../components/core-components/channel-list.mdx) component is a standalone component which (unsurprisingly) holds and manages list of channels. You can access loaded `channels` from [`ChannelListContext`](../components/contexts/channel-list-context.mdx) with the help of `useChannelListContext` hook.
+
+```tsx
+import { ChannelList, ChannelPreviewMessenger } from 'stream-chat-react';
+import type { ChannelListProps } from 'stream-chat-react';
+
+const CustomPreviewUI = (props) => {
+  const { channels } = useChannelListContext();
+
+  return <ChannelPreviewMessenger {...props} />;
+};
+
+export const CustomChannelList = (props: ChannelListProps) => {
+  return <ChannelList Preview={CustomPreviewUI} {...props} />;
+};
+```
+
+## Conclusion
+
+This guide covers the biggest and most important state stores, see other React stateful contexts exported by our SDK for more information.
```

Comment:
I think we should add at least one example of subscribing to the LLC events here, maybe as an advanced section. Otherwise it's weird that we concentrate so hard on the fact that the state is not reactive, but don't show how it works.

Reply:
I had a "to do" for this but decided against as I suspect we'd be opening an unnecessary Pandora's box for us. We have described how it works but as this is not a good pattern (at all) I'd rather just not provide a replicable source.